commit 0821d5191045306cecc0834929853c86ca085e03
Author: Dawit Alemayehu <adawit@kde.org>
Date:   Thu Sep 6 15:00:06 2012 -0400

    Do not crash when QWebHitTestResult::frame() returns NULL.
    
    BUG: 306102
    FIXED-IN: 1.3

diff --git a/src/webview.cpp b/src/webview.cpp
index e0d2407..b6ba0e8 100644
--- a/src/webview.cpp
+++ b/src/webview.cpp
@@ -547,7 +547,7 @@ void WebView::partActionPopupMenu(KParts::BrowserExtension::ActionGroupMap& part
                 partActions.append(action);
             }
         }
-    } else if (m_result.frame()->parentFrame() && !m_result.isContentSelected() && m_result.linkUrl().isEmpty()) {
+    } else if (m_result.frame() && m_result.frame()->parentFrame() && !m_result.isContentSelected() && m_result.linkUrl().isEmpty()) {
         KActionMenu * menu = new KActionMenu(i18nc("@title:menu HTML frame/iframe", "Frame"), this);
 
         KAction* action = new KAction(KIcon("window-new"), i18n("Open in New &Window"), this);
